### DISCUSSION TITLE
Disable autoplay on user input when testing a beatmap in the editor

### DISCUSSION THREAD

**Discussion by Hiviexd** [ppy/osu#28932] [Ideas]
It's intuitive behavior that's present in stable, makes switching between autoplay and actual play much smoother by removing the extra tab press when moving from the former to the latter.

**Comment by peppy** [comment]
Making this work has similar difficulties to trying to get gameplay keys to work at the pause screen. Basically we don't have a way of detecting gameplay keys from outside gameplay.

Fixing this osu!-side might look like exposing an `event` pathway via `DrawableRuleset` to map gameplay keys to normal keyboard keys.. but this seems pretty ugly. A better option may be to add more facility to `KeyBindingContainer` (the non-generic implementation) to have visibility over what keys (if any) are pressed.
